processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java

```diff
@@ -66,14 +67,12 @@ public TypeSelector(TypeFactory typeFactory, FormattingMessager messager) {
             List<List<ParameterBinding>> parameterBindingPermutations =
                 getCandidateParameterBindingPermutations( availableBindings, method.getMethod().getParameters() );
 
-            if ( parameterBindingPermutations != null ) {
```

Comment:
This makes sense from a programming perspective. However, in our case `getMatchingParameterBinding` is quite a complex method that would execute complex logic based on `parameterBindingPermutations`. Therefore I would revert the changes to the `TypeSelector` class.

Reply:
Ok